document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -81,16 +96,108 @@ public String visit(final KeyExpression expression) {
             postgresQueryParser.getParamsBuilder());
   }
 
-  @SuppressWarnings("unchecked")
+  @SuppressWarnings({"unchecked", "SwitchStatementWithTooFewBranches"})
   @Override
   public String visit(final ArrayRelationalFilterExpression expression) {
-    throw new UnsupportedOperationException();
+    /*
+    EXISTS
+     (SELECT 1
+      FROM jsonb_array_elements(COALESCE(planets->'elements', '[]'::jsonb)) AS elements
+      WHERE TRIM('"' FROM elements::text) = 'Oxygen'
+     )
+     */
+    switch (expression.getOperator()) {
+      case ANY:
```

Comment:
If there are more operators, we should move the code for each operator to its dedicated function for readability? For instance, moving code of any operator to `getFilterExpressionForAnyOperator()` here.

Reply:
Sure